docs/proposals/receiving_metadata.md

```diff
@@ -0,0 +1,49 @@
+# Support /api/v1/metadata in Cortex
+
+- Author: @gotjosh
+- Reviewers: @tomwilkie, @pracucci
+- Date: March 2020
+
+## Problem Statement
+Prometheus holds metric metadata alongside the contents of a scrape. This metadata (`HELP`, `TYPE`, `UNIT` and `METRIC_NAME`) enables [some Prometheus API](https://github.com/prometheus/prometheus/issues/6395) endpoints to output the metadata for integrations (e.g. [Grafana](https://github.com/grafana/grafana/pull/21124)) to consume it. 
+
+At the moment of writing, Cortex does not support the `api/v1/metadata` endpoint that Prometheus implements as metadata was never propagated via remote write. Recent [work is done in Prometheus](https://github.com/prometheus/prometheus/pull/6815/files) enables the propagation of metadata.
+
+With this in place, remote write integrations such as Cortex can now receive this data and implement the API endpoint. This result in Cortex users being able to enjoy a tiny bit more insight on their metrics.
+
+## Potential Solutions
+Before we delve into the solutions, let's set a baseline about how the data is received. This applies almost equally for the two.
+
+Metadata from Prometheus is sent in the same `WriteRequest` proto message that the samples use. It is part of a different field (#3 given #2 is already used), the data is a set identified by the metric name - that means it is aggregated across targets, and finally it is sent all at once. It is also important to note that this current process is an intermediary step. Eventually, metadata in a request will be sent alongside samples and only for those included. The solutions proposed, take this nuance into account to avoid coupling between the current and future state of Prometheus, and hopefully do something now that also works for the future. 
+
+As a reference, these are some key numbers regarding the size (and send timings) of the data at hand from our clusters at GL:
+
+- On average, metadata (a combination of `HELP`, `TYPE`, `UNIT` and `METRIC_NAME`) is ~55 bytes uncompressed.
+- at GL, on an instance with about 2.6M active series, we hold ~1241 unique metrics in total.
+- with that, we can assume that on a worst-case scenario the metadata set for that instance is ~68 kilobytes uncompressed.
+- by default, this data is only propagated once every minute (aligning with the default scrape interval), but this can be adjusted.
+- Finally, what this gives us is a baseline worst-case scenario formula for the data to store per tenant: `~68KB * Replication Factor * # of Instances`. Keeping in mind that typically, there's a very high overlap of metadata across instances, and we plan to deduplicate in the ingesters.
```

Comment:
I'm not sure how we can use `68KB == 1241 metrics` as worst-case scenario.

Reply:
I meant for our case, of course, YMMV depending on what metrics you have. Would you rather have me phrase this differently? e.g. remove the worse-case wording all around. 

The important comparison here is that the unique metrics to active series ratio is very very low.